core/src/main/java/org/infinispan/util/concurrent/locks/impl/PerKeyLockContainer.java

```diff
@@ -26,16 +26,19 @@
 
    private static final int INITIAL_CAPACITY = 32;
    private final ConcurrentMap<Object, InfinispanLock> lockMap;
-   private Executor executor;
+   private Executor blockingExecutor;
+   private Executor nonBlockingExecutor;
    private TimeService timeService;
 
    public PerKeyLockContainer() {
       lockMap = new ConcurrentHashMap<>(INITIAL_CAPACITY);
    }
 
    @Inject
-   void inject(@ComponentName(KnownComponentNames.BLOCKING_EXECUTOR) Executor executor, TimeService timeService) {
-      this.executor = executor;
+   void inject(@ComponentName(KnownComponentNames.BLOCKING_EXECUTOR) Executor blockingExecutor,
+         @ComponentName(KnownComponentNames.NON_BLOCKING_EXECUTOR) Executor nonBlockingExecutor, TimeService timeService) {
```

Comment:
I'd use field injection instead.

`lock.setTimeService(timeService)` looks like it might be useful, but because there is no synchronization there's always a chance of missing one of the locks, so I'd remove that and force tests to replace the time service ahead of time.

Reply:
> I'd use field injection instead.

I can change it, was just keeping it consistent with the previous code.

> `lock.setTimeService(timeService)` looks like it might be useful, but because there is no synchronization there's always a chance of missing one of the locks, so I'd remove that and force tests to replace the time service ahead of time.

I thought you said before that cache initialization was thread safe in regards to visibility? Is this not true now?

Reply:
Regular cache initialization is safe. But once a component A is started, there's no other synchronization to ensure that changes made to A's fields are properly seen by all threads that are concurrently accessing A.